src/Backend/ONE/ONEToolchain.ts

```diff
@@ -14,33 +14,148 @@
  * limitations under the License.
  */
 
+import {execSync} from 'child_process';
+import * as vscode from 'vscode';
+
 import {Backend} from '../Backend';
 import {Command} from '../Command';
 import {Compiler} from '../Compiler';
 import {Executor} from '../Executor';
-import {Toolchains} from '../Toolchain';
+import {ToolchainInfo, Toolchains} from '../Toolchain';
+import {DebianArch, DebianRepo, DebianToolchain} from '../ToolchainImpl/DebianToolchain';
+import {Version} from '../Version';
+
+
+class OneDebianToolchain extends DebianToolchain {
+  run(cfg: string): Command {
+    this.prepare();
+    let cmd = new Command('onecc-docker');
+    cmd.push('-C');
+    cmd.push(cfg);
+    return cmd;
+  }
+}
 
 class ToolchainCompiler implements Compiler {
   private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+  private readonly debianRepo: DebianRepo;
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker-test';
+    // we currently support only the latest version of one-compiler.
+    this.toolchainTypes = ['latest'];
+
+    // we temporary add personal PPA for test.
+    this.debianRepo =
+        new DebianRepo('http://ppa.launchpad.net/psalmist-kim/ppa/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
-  getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+  getVersion(version: string): Version {
```

Comment:
We decided that it was necessary to get the version information output to the terminal's stdin and convert the string version information into a Version object, so we created the getVersion() function.

We have placed the function in the current Compiler implementation, but it seems that the function getVersion() may be confused as to whether the function gets the version of the compiler.

We would like to export the function out of the class or have a suitable name, can you please give me some advice?

Reply:
How about using `parseVersion`?

Reply:
now it is `parseVersion` instead of `getVersion`.